system/jlib/jsocket.cpp

```diff
@@ -4852,6 +4856,15 @@ class CSocketEpollThread: public CSocketBaseThread
                             EXCLOG(e,"CSocketEpollThread notifySelected");
                             throw ;
                         }
+                        try
+                        {
+                            si.nfy->Release();
```

Comment:
If this throw an exception then it would cause sock to leak.
But neither of these should really throw excepetion in their destructor..
Have you looked if they can and considered catching any exception they might in their destructors?